bin/jekyll-auth

```diff
@@ -1,179 +1,181 @@
+    c.option "app", "--app", "The name of an existing Heroku app to use"
+    c.option "client_id", "--client_id", "Your oauth app client id"
+    c.option "client_secret", "--client_secret", "Your oauth app client secret"
+    c.option "team_id", "--team_id", "The team to authenticate against"
+    c.option "org_id", "--org_id", "An organization to authenticate against"
+    c.action do |args, options|
+
+      if find_executable("heroku").nil?
+        say "Looks like we're missing the Heroku client. Let's see if we can't install it..."
+        execute_command "wget", "-qO-", "https://toolbelt.heroku.com/install.sh", "|", "sh"
+      end
 
-    # build site
-    command(:build).run
+      git = Git.init
+      git.add ".gitignore"
+      git.add "config.ru"
+      git.add "Rakefile"
 
-    say "Spinning up the server with authentication. Use CTRL-C to stop."
-    say "To preview the site without authentication, use the `jekyll serve` command"
-    sh "bundle exec rackup -p 4000"
+      if changed?
+        git.commit "[Jekyll Auth] Initial setup"
+      end
 
-  end
-end
+      if git.remotes.any? { |remote| remote.name == "heroku" }
+        puts "Looks like you've already got heroku set up... skipping.".green
+      else
+        if options["app"]
+          puts "Creating a new Heroku app."
+          execute_command "heroku", "create"
+        else
+          puts "Telling Heroku to use your existing app."
+          execute_command "heroku", "git:remote", "-a", options["app"]
+        end
+      end
 
-# Run the standard jekyll build command
-# Called by Rake task, to allow the gem
-# to add functionality here in the future
-command :build do |c|
-  c.syntax = 'jekyll-auth build'
-  c.description = "Build Jekyll site"
-  c.action do |args, options|
-    say "building the site..."
-    sh "bundle exec jekyll build"
-    say "site built."
-  end
-end
+      puts "Configuring the Heroku app"
+      execute_command "heroku", "config:set", "GITHUB_CLIENT_ID=#{options["client_id"]}", "GITHUB_CLIENT_SECRET=#{options["client_secret"]}"
 
-command :team_id do |c|
-  c.syntax = 'jekyll-auth team_id --org <ORG> --team <TEAM>'
-  c.description = "Retrieve a team's ID"
-  c.option '--org <ORG>', String, 'The GitHub Organization, e.g., "jekyll"'
-  c.option '--team <TEAM>', String, 'The team name, e.g., "maintainers"'
-
-  c.action do |args, options|
-    if ENV["GITHUB_TOKEN"].to_s.blank?
-      say "You'll need to go to https://github.com/settings/tokens/new and create a personal access token"
-      say "Once you've got the token, prefix the jekyll-auth command with GITHUB_TOKEN=[YOUR TOKEN]"
-      say "You can also add it to a `.env` file in this directory"
-      exit 1
-    end
+      if options["team_id"]
+        execute_command "heroku", "config:set", "GITHUB_TEAM_ID=#{options["team_id"]}"
+      else
+        execute_command "heroku", "config:set", "GITHUB_ORG_ID=#{options["org_id"]}"
+      end
```

Comment:
This is dicey. I'm using the `hk` go executable and don't have `heroku` installed. `hk` uses `set` instead of `config:set`. This will fail if I don't have heroku installed and will doubly feel bad when I don't want to use `heroku` but am being asked to anyway.